src/main/core/logger/log_wrapper.c

```diff
@@ -0,0 +1,29 @@
+#include "main/core/logger/log_wrapper.h"
+
+#include "main/bindings/c/bindings.h"
+#include "support/logger/logger.h"
+
+static void _log(Logger* logger, LogLevel level, const char* fileName, const char* functionName,
+                 const int lineNumber, const char* format, va_list vargs) {
+    rustlogger_log(level, fileName, functionName, lineNumber, format, vargs);
+}
+
+static void _flush(Logger* logger) { rustlogger_flush(); }
+
+static bool _isEnabled(Logger* logger, LogLevel level) { return rustlogger_isEnabled(level); }
+
+static void _setLevel(Logger* logger, LogLevel level) { rustlogger_setLevel(level); }
+
+Logger* rustlogger_new() {
+    Logger* logger = malloc(sizeof(*logger));
+    *logger = (Logger){
+        .log = _log,
+        .flush = _flush,
+        .destroy = rustlogger_destroy,
+        .isEnabled = _isEnabled,
+        .setLevel = _setLevel,
+    };
+    return logger;
+}
+
+void rustlogger_destroy(Logger* logger) { free(logger); }
```

Comment:
FYI: I've recently come to appreciate files ending with a blank line rather than the last line not containing a newline. When adding new lines in the future, the diff will show that the previous last line in the file is modified even though the only thing changed was the addition of a newline character.

Reply:
Done. Agreed; I'm used to vim doing it automatically. I'll see if there's a VSCode setting...